app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/DefaultTransactionInfoRepository.kt

```diff
@@ -88,18 +93,41 @@ class DefaultTransactionInfoRepository @Inject constructor(
                 TransactionData.Generic(address, value?.value ?: BigInteger.ZERO, data, Operation.CALL)
         }
 
-    // TODO: This need to be adjusted for the new MultiSend
     private fun isMultiSend(safeTransaction: SafeTransaction) =
         safeTransaction.operation == Operation.DELEGATE_CALL &&
-                safeTransaction.wrapped.address == MULTI_SEND_LIB &&
                 safeTransaction.wrapped.data != null &&
                 safeTransaction.wrapped.data!!.isSolidityMethod(MultiSend.MultiSend.METHOD_ID)
 
-    // TODO: This need to be adjusted for the new MultiSend
-    private fun parseMultiSend(transaction: SafeTransaction): TransactionData.MultiSend {
+    private fun parseMultiSend(transaction: SafeTransaction): TransactionData {
         val payload =
-            transaction.wrapped.data?.removeSolidityMethodPrefix(MultiSend.MultiSend.METHOD_ID) ?: return TransactionData.MultiSend(emptyList())
+            transaction.wrapped.data?.removeSolidityMethodPrefix(MultiSend.MultiSend.METHOD_ID)
+                ?: return TransactionData.MultiSend(emptyList(), transaction.wrapped.address)
+
+        val multiSend = when (transaction.wrapped.address) {
+            MULTI_SEND_LIB -> nullOnThrow { parseMultiSendNew(payload) }
```

Comment:
MultiSendNew won't be new over time. Just MultiSend would be better naming in my opinion